File: src/main/java/com/SpringBootProject/serviceInterface/UserService.java
package com.SpringBootProject.serviceInterface;

import java.security.NoSuchAlgorithmException;
import java.util.List;

import com.SpringBootProject.requestDto.UserRequestDto;
import com.SpringBootProject.responseDto.LoginResponseDto;
import com.SpringBootProject.responseDto.ResponseDto;

public interface UserService {

	List<LoginResponseDto> getAllUsers();
	LoginResponseDto getUserById(long userId);
	ResponseDto addNewUser(UserRequestDto userData) throws NoSuchAlgorithmException;
	ResponseDto updateUser(long userId,UserRequestDto userData) throws NoSuchAlgorithmException;
}

File: DP_UI/dp-frontend/src/service/UserService.js
import { myAxios } from "./helper"

class UserService{
    getAllUsers(){
        return myAxios
            .get('/users')
            .then((response) => response.data)
    }

    addNewUser(userData){
        return myAxios
            .post("/users",userData)
            .then((response) => response.data)
    }

    getUserById(userId){
        return myAxios
            .get("/users/"+userId)
            .then((response) => response.data)
    }

    editUser(editedUserData,userId){
        return myAxios
            .put("/users/"+userId,editedUserData)
            .then((response) => response.data)
    }
}

export default new UserService();